HW3/P4/median_filter.cl

```diff
@@ -12,10 +12,56 @@ median_3x3(__global __read_only float *in_values,
     // Note: It may be easier for you to implement median filtering
     // without using the local buffer, first, then adjust your code to
     // use such a buffer after you have that working.
+    const int x = get_global_id(0);
+    const int y = get_global_id(1);
 
+    const int lx = get_local_id(0);
+    const int ly = get_local_id(1);
 
+    const int buf_corner_x = x - lx - halo;
+    const int buf_corner_y = y - ly - halo;
+
+    const int buf_x = lx + halo;
+    const int buf_y = ly + halo;
+
+    const int idx_1D = ly * get_local_size(0) + lx;
+
+     int row, tx, ty;
+
+     if (idx_1D < buf_w){
+         for (row = 0; row < buf_h; row++) {
+           tx = idx_1D;
+           ty = row;
+          if (tx+buf_corner_x < 0){
+            tx++;
+          } else if(tx+buf_corner_y == w) {
+            tx--;
```

Comment:
This may cause an out-of-bounds access to the original input data, because the buffer on the right-most work group might have more than 1 column to the right of the input matrix.